test/document_stores/test_memory.py

```diff
@@ -124,3 +126,19 @@ def test_memory_query_by_embedding_docs_wo_embeddings(self, ds, caplog):
             docs = ds.query_by_embedding(query_emb=query_embedding, top_k=1)
             assert "Skipping some of your documents that don't have embeddings" in caplog.text
         assert len(docs) == 0
+
+    @pytest.mark.integration
+    def test_bm25_scores_not_changing_across_queries(self, ds, documents):
+        """Test that computed scores which are returned to the user should not change when running multiple queries."""
+        ds.write_documents(documents)
+        retriever = BM25Retriever(ds, scale_score=False)
+        queries = ["What is a Foo Document?", "What is a Bar Document?", "Tell me about a document without embeddings"]
+        results_direct = []
+        for query in queries:
+            documents_with_scores = retriever.retrieve(query)
+            results_direct.append(documents_with_scores)
+        results_copied = [deepcopy(retriever.retrieve(query)) for query in queries]
```

Comment:
Let's make them look equal, so it's more obvious what is being tested here:

```suggestion
        results_direct = [retriever.retrieve(query) for query in queries]
        results_copied = [deepcopy(retriever.retrieve(query)) for query in queries]
```

Reply:
Thank makes sense. Committed your suggestion.